t3391: Treat non-critical runtime deployment timeout as setup warning

## What
Fix setup.sh so a non-critical deploy_agents_to_runtimes timeout in non-interactive setup does not make the whole setup command exit 1 after core deployment has already succeeded.

## Why
During postflight for GH#22138 after release v3.13.67, ./setup.sh --non-interactive completed agent deployment and pulse restart, then logged: Runtime agent deployment exceeded 120s — skipping (non-critical), Finished setup stage: deploy_agents_to_runtimes (122.33s, exit=1), [ERROR] ./setup.sh:1380 exit 1. A stage explicitly described as non-critical should not fail the lifecycle postflight unless setup intentionally treats it as fatal.

## Files to modify
- EDIT: setup.sh — inspect the run_stage/deploy_agents_to_runtimes non-interactive path around the failing call site reported at setup.sh:1380.
- EDIT: .agents/scripts/tests/ or existing setup tests — add a regression proving the non-critical runtime deployment timeout path exits successfully or is otherwise classified consistently.

## Reference pattern
Model on existing setup non-interactive lock and timeout tests in .agents/scripts/tests/test-setup-noninteractive-lock.sh and existing setup stage handling patterns in setup.sh.

## Verification
- Run the new/updated setup regression test.
- Run shellcheck on setup.sh and edited tests.
- Run ./setup.sh --non-interactive in an environment that exercises or stubs the runtime deployment timeout path and confirm it does not fail after logging non-critical skip.

## Brief Workflow

This issue body is composed under `.agents/workflows/brief.md`. Before implementation, ensure it contains files to modify, a reference pattern, verification commands, and concrete acceptance criteria.

<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.13.67 plugin for [OpenCode](https://opencode.ai) v1.14.31 with gpt-5.5 spent 59m and 182,215 tokens on this as a headless worker.

--- a/.agents/scripts/tests/test-deploy-runtimes-bounded.sh
+++ b/.agents/scripts/tests/test-deploy-runtimes-bounded.sh
@@ -220,6 +220,49 @@ test_bounded_wrapper_timeout_env_respected() {
 	return 0
 }
 
+test_real_bounded_wrapper_timeout_returns_zero() {
+	TEST_TMP_DIR="$(mktemp -d)"
+	local script="${TEST_TMP_DIR}/real_wrapper_timeout.sh"
+	local wrapper_definition=""
+
+	wrapper_definition=$(awk '
+		/^_deploy_agents_to_runtimes_bounded\(\)/{found=1; depth=0}
+		found {print}
+		found && /\{/{depth++}
+		found && /\}/{depth--; if(depth<=0){exit}}
+	' "$AGENT_RUNTIME_SH")
+
+	if [[ -z "$wrapper_definition" ]]; then
+		print_result "real bounded wrapper extraction succeeds" 1 "function not found in ${AGENT_RUNTIME_SH}"
+		return 0
+	fi
+
+	{
+		printf '%s\n' '#!/usr/bin/env bash'
+		printf '%s\n' 'set -uo pipefail'
+		printf '%s\n' 'print_warning() { printf "%s\n" "$*"; return 0; }'
+		printf '%s\n' 'deploy_agents_to_runtimes() { sleep 10; return 0; }'
+		printf '%s\n' 'AIDEVOPS_DEPLOY_RUNTIMES_TIMEOUT=2'
+		printf '%s\n' "$wrapper_definition"
+		printf '%s\n' '_deploy_agents_to_runtimes_bounded'
+	} >"$script"
+	chmod +x "$script"
+
+	local output=""
+	local rc=0
+	local start_s=$SECONDS
+	output=$(bash "$script" 2>&1) || rc=$?
+	local elapsed=$(( SECONDS - start_s ))
+
+	if [[ "$rc" -eq 0 && "$elapsed" -le 7 && "$output" == *"non-critical"* ]]; then
+		print_result "real bounded wrapper timeout returns zero with non-critical warning" 0
+	else
+		print_result "real bounded wrapper timeout returns zero with non-critical warning" 1 \
+			"rc=$rc elapsed=${elapsed}s output=${output}"
+	fi
+	return 0
+}
+
 main() {
 	trap cleanup EXIT
 
@@ -231,6 +274,7 @@ main() {
 	test_bounded_wrapper_fast_path
 	test_bounded_wrapper_kills_slow_deployment_without_failing_setup
 	test_bounded_wrapper_timeout_env_respected
+	test_real_bounded_wrapper_timeout_returns_zero
 
 	printf '\nRan %s tests, %s failed\n' "$TESTS_RUN" "$TESTS_FAILED"
 	if [[ "$TESTS_FAILED" -ne 0 ]]; then

--- a/.agents/scripts/tests/test-deploy-runtimes-bounded.sh
+++ b/.agents/scripts/tests/test-deploy-runtimes-bounded.sh
@@ -250,15 +250,13 @@ test_real_bounded_wrapper_timeout_returns_zero() {
 
 	local output=""
 	local rc=0
-	local start_s=$SECONDS
 	output=$(bash "$script" 2>&1) || rc=$?
-	local elapsed=$(( SECONDS - start_s ))
 
-	if [[ "$rc" -eq 0 && "$elapsed" -le 7 && "$output" == *"non-critical"* ]]; then
+	if [[ "$rc" -eq 0 && "$output" == *"non-critical"* ]]; then
 		print_result "real bounded wrapper timeout returns zero with non-critical warning" 0
 	else
 		print_result "real bounded wrapper timeout returns zero with non-critical warning" 1 \
-			"rc=$rc elapsed=${elapsed}s output=${output}"
+			"rc=$rc output=${output}"
 	fi
 	return 0
 }

--- a/.agents/scripts/tests/test-deploy-runtimes-bounded.sh
+++ b/.agents/scripts/tests/test-deploy-runtimes-bounded.sh
@@ -6,7 +6,7 @@
 # Verifies that the bounded wrapper:
 #   1. Returns 0 when deploy_agents_to_runtimes completes quickly.
 #   2. Kills a slow deployment within AIDEVOPS_DEPLOY_RUNTIMES_TIMEOUT seconds and
-#      returns non-zero, ensuring the stage cannot hang the outer setup timeout.
+#      returns 0 because runtime deployment is non-critical during setup postflight.
 #   3. The bounded wrapper is wired into setup.sh's non-interactive path.
 #
 # Note on process-group cleanup: deploy_agents_to_runtimes does not start any
@@ -89,7 +89,7 @@ _deploy_agents_to_runtimes_bounded() {
 			kill -KILL "\$_pid" 2>/dev/null || true
 			wait "\$_pid" 2>/dev/null || true
 			print_warning "bounded: exceeded timeout"
-			return 1
+			return 0
 		fi
 		sleep 1
 	done
@@ -170,7 +170,7 @@ test_bounded_wrapper_fast_path() {
 	return 0
 }
 
-test_bounded_wrapper_kills_slow_deployment() {
+test_bounded_wrapper_kills_slow_deployment_without_failing_setup() {
 	TEST_TMP_DIR="$(mktemp -d)"
 	local script="${TEST_TMP_DIR}/slow.sh"
 
@@ -181,11 +181,11 @@ test_bounded_wrapper_kills_slow_deployment() {
 	bash "$script" || rc=$?
 	local elapsed=$(( SECONDS - start_s ))
 
-	if [[ "$rc" -ne 0 ]]; then
-		print_result "slow deployment: bounded wrapper returns non-zero on timeout" 0
+	if [[ "$rc" -eq 0 ]]; then
+		print_result "slow deployment timeout is non-critical for setup" 0
 	else
-		print_result "slow deployment: bounded wrapper returns non-zero on timeout" 1 \
-			"expected non-zero from bounded wrapper, got 0"
+		print_result "slow deployment timeout is non-critical for setup" 1 \
+			"expected 0 from bounded wrapper timeout, got $rc"
 	fi
 
 	# timeout=3 + 2s SIGKILL grace + 1s poll overhead = at most ~7s
@@ -203,6 +203,7 @@ test_bounded_wrapper_timeout_env_respected() {
 	local script="${TEST_TMP_DIR}/env_check.sh"
 
 	# Deploy that sleeps 10s; AIDEVOPS_DEPLOY_RUNTIMES_TIMEOUT=2 should terminate it
+	# without failing setup's non-interactive path.
 	_write_bounded_wrapper_script "$script" "  sleep 10; return 0" "2"
 
 	local rc=0
@@ -210,7 +211,7 @@ test_bounded_wrapper_timeout_env_respected() {
 	bash "$script" || rc=$?
 	local elapsed=$(( SECONDS - start_s ))
 
-	if [[ "$rc" -ne 0 && "$elapsed" -le 7 ]]; then
+	if [[ "$rc" -eq 0 && "$elapsed" -le 7 ]]; then
 		print_result "AIDEVOPS_DEPLOY_RUNTIMES_TIMEOUT env var respected (${elapsed}s)" 0
 	else
 		print_result "AIDEVOPS_DEPLOY_RUNTIMES_TIMEOUT env var respected (${elapsed}s)" 1 \
@@ -228,7 +229,7 @@ main() {
 	test_setup_uses_bounded_wrapper
 	test_deploy_function_has_no_background_jobs
 	test_bounded_wrapper_fast_path
-	test_bounded_wrapper_kills_slow_deployment
+	test_bounded_wrapper_kills_slow_deployment_without_failing_setup
 	test_bounded_wrapper_timeout_env_respected
 
 	printf '\nRan %s tests, %s failed\n' "$TESTS_RUN" "$TESTS_FAILED"

--- a/setup-modules/agent-runtime.sh
+++ b/setup-modules/agent-runtime.sh
@@ -256,7 +256,8 @@ deploy_agents_to_runtimes() {
 # cannot be passed directly to the `timeout` binary. Instead we run it in a
 # background subshell and poll for completion with a configurable wall-clock
 # deadline. When the deadline expires the subshell is killed with SIGTERM then
-# SIGKILL and setup continues without this non-critical step.
+# SIGKILL and this non-critical step is treated as a warning so setup can
+# continue after core deployment has already succeeded.
 #
 # Configurable via AIDEVOPS_DEPLOY_RUNTIMES_TIMEOUT (default 120s).
 _deploy_agents_to_runtimes_bounded() {
@@ -275,7 +276,7 @@ _deploy_agents_to_runtimes_bounded() {
 			kill -KILL "$_pid" 2>/dev/null || true
 			wait "$_pid" 2>/dev/null || true
 			print_warning "Runtime agent deployment exceeded ${timeout_s}s — skipping (non-critical)"
-			return 1
+			return 0
 		fi
 		sleep 1
 	done

